File: leetcode454/leetcode454.cpp
#include<bits/stdc++.h>
using namespace std;
class Solution {
    public:
        int fourSumCount(vector<int>& nums1, vector<int>& nums2, vector<int>& nums3, vector<int>& nums4) {
            unordered_map<int,int> map;
            int res = 0;
            for(auto a : nums1){
                for(auto b : nums2){
                    map[a + b]++;
                }
            }
            for(auto a : nums3){
                for(auto b : nums4){
                    auto iter = map.find(0 - a - b);
                    if(iter != map.end()){
                        res += map[0 - a - b]; 
                    }
                }
            }
            return res;
        }
    };

File: leetcode454/leetcode454.py
class Solution:
    def fourSumCount(self, nums1: List[int], nums2: List[int], nums3: List[int], nums4: List[int]) -> int:
        hash = dict()
        res = 0
        for i in nums1:
            for j in nums2:
                hash[i + j] = hash.get(i + j, 0) + 1
        for i in nums3:
            for j in nums4:
                if 0 - i - j in hash:
                    res += hash[0 - i - j]
        return res